### PR TITLE
Drusu/vcbc figure 2 no state

### PR DESCRIPTION
**Pull request by davidrusu**

### CODE REVIEW

File: src/mvba/vcbc/mod.rs [hunk 1]
```diff
@@ -1,75 +1,249 @@
...
+
+                // if H(m̄) = d and µ̄ = ⊥ and µ is a valid S1 -signature then
+                if d == &msg_d && self.u_bar.is_none() && valid_sig {
+                    // µ̄ ← µ
+                    self.u_bar = Some(sig);
+                }
+            }
+        }
+
+        for (sender, final_msg) in std::mem::take(&mut self.final_messages) {
```
Comment (b00f) [inline src/mvba/vcbc/mod.rs:197]: I prefer keep all messages in buffer and then process them. Other wise we need put some comments here to explain why we only buffered the final messages.

File: src/mvba/vcbc/tests.rs [hunk 2]
```diff
@@ -1,174 +1,379 @@
...
+            recipient_node.receive_message(bundle.sender, msg);
+            self.enqueue_bundles_from(recipient);
+        }
     }
 }
 
 #[test]
-fn test_should_propose() {
-    let mut t = TestData::new(TestData::PARTY_X);
+fn test_vcbc_happy_path() {
```
Comment (b00f) [inline src/mvba/vcbc/tests.rs:124]: This looks like a system test for me, not a unit test. In unit tests we should test as much as possible a small unit of the code.

**Review by b00f — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -37,3 +37,4 @@ eyre = "0.6.5"
 quickcheck = "1"
 quickcheck_macros = "1"
 env_logger = "0.8"
+hex = "0.4"
```
Comment (b00f) [inline Cargo.toml:40]: Remove this dependency, It is only used in one tests.

File: src/mvba/vcbc/tests.rs (hunk 2 above)
Comment (davidrusu) [reply src/mvba/vcbc/tests.rs:124]: Discussed in syncup: will keep these system tests here in this file until the full implementation is ready to replace the sn_consensus crate

File: src/mvba/vcbc/mod.rs [hunk 4]
```diff
@@ -1,75 +1,249 @@
...
+                    // if i = j and νl is a valid S1-signature share then
+                    if self.i == msg.tag.j && valid_sig {
+                        // Wd ← Wd ∪ {νl}
+                        e.insert(sig_share.clone());
+
+                        //  rd ← rd + 1
+                        self.rd += 1;
+
+                        // if rd = n+t+1/2 then
+                        if self.rd > self.threshold() {
```
[the remark this answers is not in the document]
Comment (davidrusu) [reply src/mvba/vcbc/mod.rs:156]: Discussed in syncup: It would be difficult to keep track of the `t` in safe network making future integration difficult.

**Pull request merged**